internal/core/command/router.go

```diff
@@ -103,7 +103,8 @@ func loadDeviceRoutes(b *mux.Router, dic *di.Container) {
 				bootstrapContainer.LoggingClientFrom(dic.Get),
 				container.DBClientFrom(dic.Get),
 				commandContainer.MetadataDeviceClientFrom(dic.Get),
-				errorContainer.ErrorHandlerFrom(dic.Get))
+				errorContainer.ErrorHandlerFrom(dic.Get),
+				&http.Client{})
```

Comment:
Would it be better to put this into the dependency injection container(dic) and retrieve the value like we are doing for the other components? However, I am wondering if this would create a race condition since we can only have one instance of `http.Client` in the `dic` and each time this REST handler is invoked it would be on a different go-routine and each of them would be using the same instance of `http.Client` and some of the methods look like there is some state being accessed for each call. Any thoughts?

Reply:
Maybe we can add a comment in the code stating that so our thoughts are documented.